IBP-1971 Exclude null parent gid

gelli-leafnode opened the pull request:
Catch Middleware exception and replace to recommended error message

gelli-leafnode committed:
Catch MiddlewareException for Pedigree error and showing recommended error message.
IBP-1971-ExcludeNullParentGid

gelli-leafnode committed:
Catch MiddlewareException for Pedigree error and showing recommended error message.
IBP-1971-ExcludeNullParentGid

gelli-leafnode committed:
Add condition to show MessageNotifier if error in build pedigree string.
IBP-1971-ExcludeNullParentGid

gelli-leafnode committed:
Code clean up.
IBP-1971-ExcludeNullParentGid

gelli-leafnode committed:
Revert set static for TableRightClickHandler suggested by code analyzer
IBP-1971-ExcludeNullParentGid

gelli-leafnode committed:
Revert set set final for listmanager.
IBP-1971-ExcludeNullParentGid

darla-leafnode commented on src/main/java/org/generationcp/breeding/manager/listmanager/GermplasmSearchBarComponent.java:
```diff
@@ -268,27 +259,31 @@ public void updateLabels() {
+				GermplasmSearchBarComponent.this.doSearch(q);
+			}
+		}catch(Exception ex) {
+			LOG.debug(ex.getMessage());
```
Why do we need this change? It seems to just be logging in debug mode

darla-leafnode commented on src/main/java/org/generationcp/breeding/manager/listmanager/listeners/FillWithMenuItemClickListener.java:
```diff
@@ -144,8 +146,13 @@ private void displayExpansionLevelPopupWindow(final String columnName) {
+				try{
+					FillWithMenuItemClickListener.this.valuesGenerator.fillWithCrossExpansion(crossExpansionLevel,
+							columnName);
+				}catch (MiddlewareException ex){
```
Code is not formatted. Add final to` MiddlewareException ex`

darla-leafnode commented on src/main/java/org/generationcp/breeding/manager/listmanager/GermplasmSearchResultsComponent.java:
```diff
@@ -260,11 +261,20 @@ public String generateDescription(final Component source, final Object itemId, f
+								Integer.parseInt(((Button) item.getItemProperty(ColumnLabels.GID.getName()).getValue()).getCaption());
+
+						germplasmNames = GermplasmSearchResultsComponent.this.getGermplasmNames(gid);
+					}catch(MiddlewareException ex){
```
Add final to `MiddlewareException ex`. Apply formatter

darla-leafnode commented on src/main/java/org/generationcp/breeding/manager/listmanager/GermplasmSearchResultsComponent.java:
```diff
@@ -260,11 +261,20 @@ public String generateDescription(final Component source, final Object itemId, f
+
+						germplasmNames = GermplasmSearchResultsComponent.this.getGermplasmNames(gid);
+					}catch(MiddlewareException ex){
+						if(ex.getMessage().contains("Problem building pedigree string for gid")){
```
Maybe extract this as constant in PedigreeServiceImpl: `Problem building pedigree string for gid` to be referenced here

darla-leafnode replied:
You're swallowing the exception here for the case of non-cross expansion related exception. 
2 options:
1. Re-throw the exception
2. Display error message containing the exception's message

darla-leafnode commented:
We don't have a ready test harness for these classes to be able to create unit tests?

darla-leafnode commented on src/main/java/org/generationcp/breeding/manager/listmanager/GermplasmSearchResultsComponent.java:
```diff
@@ -260,11 +261,20 @@ public String generateDescription(final Component source, final Object itemId, f
+						germplasmNames = GermplasmSearchResultsComponent.this.getGermplasmNames(gid);
+					}catch(MiddlewareException ex){
+						if(ex.getMessage().contains("Problem building pedigree string for gid")){
+							MessageNotifier.showError(GermplasmSearchResultsComponent.this.getWindow(), "Error with Cross Expansion", "There is a data problem that prevents the generation of the cross expansion. Please contact your administrator");
```
Exception  swallowing happening here. As discussed, this is not the best place to handle MiddlewareException.

darla-leafnode committed:
Merge branch 'master' into IBP-1971-ExcludeNullParentGid

gelli-leafnode committed:
Merge branch 'master' of https://github.com/IntegratedBreedingPlatform/BreedingManager into IBP-1971-ExcludeNullParentGid

gelli-leafnode committed:
Merge branch 'IBP-1971-ExcludeNullParentGid' of https://github.com/IntegratedBreedingPlatform/BreedingManager into IBP-1971-ExcludeNullParentGid

Commit:
Merge branch 'master' of https://github.com/IntegratedBreedingPlatform/BreedingManager into IBP-1971-ExcludeNullParentGid

gelli-leafnode committed:
Method for validatePedigree to prevent creating table rows if data has error in crossing
Move showError to actual callback to resolve issue: intermittent display of error message
Catch middlewareexception in setcontainerdatasource
IBP-1971-ExcludeNullParentGid

gelli-leafnode committed:
Code cleanup
IBP-1971-ExcludeNullParentGid

gelli-leafnode committed:
Fix failing unit test: remove final from listManagerMain variable
IBP-1971-ExcludeNullParentGid

gelli-leafnode committed:
Catch exception on match containing search germplasm
IBP-1971-ExcludeNullParentGid

gelli-leafnode committed:
Include GID in error message.
IBP-1971-ExcludeNullParentGid

gelli-leafnode committed:
Include GID in error message when fillwith
IBP-1971-ExcludeNullParentGid

gelli-leafnode committed:
Code cleanup
IBP-1971-ExcludeNullParentGid
